src/index.js

```diff
@@ -105,12 +105,23 @@ class LevelDatastore {
       values = !q.keysOnly
     }
 
+    const opts = {
+      keys: true,
+      values: values,
+      keyAsBuffer: true
+    }
+
+    // Let the db do the prefix matching
+    if (q.prefix != null) {
+      const prefix = q.prefix.toString()
+      // Match keys greater than or equal to `prefix` and
+      opts.gte = prefix
+      // less than `prefix` + \xFF (hex escape sequence)
+      opts.lt = prefix + '\xFF'
+    }
+
     let it = levelIteratorToIterator(
-      this.db.db.iterator({
-        keys: true,
-        values: values,
-        keyAsBuffer: true
-      })
+      this.db.db.iterator(opts)
```

Comment:
May also want to use `this.db.iterator(opts)`, assuming that `this.db` is an instance of `levelup` >= 3.1.0. It does [exactly the same](https://github.com/Level/levelup/blob/1bbfdce6abeb6be83dc9bd37fbaab8210301e787/lib/levelup.js#L298-L300) in practice but avoids reaching into internals here.

Reply:
Ah good call. I'll add that, I think that's a safe assumption at this stage right?